betka.py

```diff
@@ -0,0 +1,244 @@
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import os
+import subprocess
+import sys
+import shutil
+from typing import List
+from pathlib import Path
+from contextlib import contextmanager
+from os import getenv
+from tempfile import TemporaryDirectory
+from collections import namedtuple
+
+
+def run_cmd(cmd, return_output=False, ignore_error=False, shell=False, **kwargs):
+    """
+    Run provided command on host system using the same user as invoked this code.
+    Raises subprocess.CalledProcessError if it fails.
+    :param cmd: list or str
+    :param return_output: bool, return output of the command
+    :param ignore_error: bool, do not fail in case nonzero return code
+    :param shell: bool, run command in shell
+    :param kwargs: pass keyword arguments to subprocess.check_* functions; for more info,
+            please check `help(subprocess.Popen)`
+    :return: None or str
+    """
+    print(f"command: {cmd}")
+    try:
+        if return_output:
+            return subprocess.check_output(
+                cmd,
+                stderr=subprocess.STDOUT,
+                universal_newlines=True,
+                shell=shell,
+                **kwargs,
+            )
+        else:
+            return subprocess.check_call(cmd, shell=shell, **kwargs)
+    except subprocess.CalledProcessError as cpe:
+        if ignore_error:
+            if return_output:
+                return cpe.output
+            else:
+                return cpe.returncode
+        else:
+            print(f"ERROR: failed with code {cpe.returncode} and output:\n{cpe.output}")
+            raise cpe
+
+
+@contextmanager
+def cwd(target):
+    """
+    Manage cwd in a pushd/popd fashion.
+    Usage:
+    with cwd(tmpdir):
+      do something in tmpdir
+    """
+    curdir = os.getcwd()
+    os.chdir(target)
+    try:
+        yield
+    finally:
+        os.chdir(curdir)
+
+
+class BetkaGenerator(object):
+
+    def __init__(self):
+        self.betka_tmp_dir = None
+        self.os_env: str = getenv("OS")
```

Comment:
I'd use `os.getenv` like you already do for the rest of the functions from this module.
```suggestion
        self.os_env: str = os.getenv("OS")
```
Then, you can remove `from os import getenv`